shared-bindings/displayio/Group.c

```diff
@@ -41,57 +41,236 @@
 //|
 //| Manage a group of sprites and groups and how they are inter-related.
 //|
-//| .. warning:: This will be changed before 4.0.0. Consider it very experimental.
-//|
-//| .. class:: Group(*, max_size=4)
+//| .. class:: Group(*, max_size=4, scale=1)
 //|
 //|   Create a Group of a given size.
 //|
 //|   :param int max_size: The maximum group size.
 //|
 STATIC mp_obj_t displayio_group_make_new(const mp_obj_type_t *type, size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
-    enum { ARG_max_size };
+    enum { ARG_max_size, ARG_scale };
     static const mp_arg_t allowed_args[] = {
         { MP_QSTR_max_size, MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = 4} },
+        { MP_QSTR_scale, MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = 1} },
     };
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
 
     mp_int_t max_size = args[ARG_max_size].u_int;
     if (max_size < 1) {
-        mp_raise_ValueError(translate("Group must have size at least 1"));
+        mp_raise_ValueError_varg(translate("Group must have %q at least 1"), MP_QSTR_max_size);
+    }
+
+    mp_int_t scale = args[ARG_scale].u_int;
+    if (scale < 1) {
+        mp_raise_ValueError_varg(translate("Group must have %q at least 1"), MP_QSTR_scale);
     }
 
     displayio_group_t *self = m_new_obj(displayio_group_t);
     self->base.type = &displayio_group_type;
-    common_hal_displayio_group_construct(self, max_size);
+    common_hal_displayio_group_construct(self, max_size, scale);
 
     return MP_OBJ_FROM_PTR(self);
 }
 
+// Helper to ensure we have the native super class instead of a subclass.
+static displayio_group_t* native_group(mp_obj_t group_obj) {
+    mp_obj_t native_group = mp_instance_cast_to_native_base(group_obj, &displayio_group_type);
+    assert(native_group == MP_OBJ_NULL);
```

Comment:
Does this raise a Python exception if it fails?

Reply:
I don't think it does anything. Seems like it should fail. Removed.